charts/gateway-helm/README.md

```diff
@@ -55,6 +59,16 @@ To uninstall the chart:
     helm delete eg
 ```
 
+### Certgen
+
+Certgen is used to generate the certificates required by EnvoyGateway.
+
+Envoy Gateway establishes a secure TLS connection for control plane communication between Envoy Gateway pods and the Envoy Proxy fleet. The TLS Certificates used here are self signed and generated using a job that runs before envoy gateway is created, and these certs and mounted on to the envoy gateway and envoy proxy pods.
+
+If you want to construct a custom certificate, you can generate a custom certificate through Cert-Manager before installing EnvoyGateway. Certgen will not overwrite the custom certificate.
+
+> _WARNING:_ Please do not manually modify `values.yaml` to disable certgen, which may cause EnvoyGateway OIDC to not work as expected.
```

Comment:
We don't need to emphasize this cause there's no configuration knob in values.yaml file that disables certgen.

Reply:
I think we need to emphasize this: this is to prevent users from unknowingly disabling certgen functionality by manually removing the certgen Job.
We still need to keep it, but it is definitely not controlled by `values.yaml`, I will change this